internal/injector/builtin/yaml/rpc/twirp.yml

```diff
@@ -0,0 +1,37 @@
+# Unless explicitly stated otherwise all files in this repository are licensed
+# under the Apache License Version 2.0.
+# This product includes software developed at Datadog (https://www.datadoghq.com/).
+# Copyright 2023-present Datadog, Inc.
+---
+# yaml-language-server: $schema=../../../../../docs/static/schema.json
+meta:
+  name: github.com/twitchtv/twirp
+  description: A simple RPC framework with protobuf service definitions.
+  icon: at-symbol
+
+aspects:
+  # TODO: Client Instrumentation - dd-trace-go does not have a *twirp.ClientHooks implementation for now.
+
+  # Server Instrumentation
+  - id: Add twirp server interceptors
+    join-point:
+      struct-literal:
+        type: github.com/twitchtv/twirp.ServerOptions
+        match: any
+    advice:
+      - wrap-expression:
+          imports:
+            twirptrace: gopkg.in/DataDog/dd-trace-go.v1/contrib/twitchtv/twirp
+          template: |-
+            {{- .AST.Type -}}{
+              {{- $hasField := false -}}
+              {{ range .AST.Elts }}
+              {{- if eq .Key.Name "Hooks" }}
+              {{- $hasField = true -}}
+              {{- end -}}
+              {{ . }},
+              {{ end -}}
+              {{- if not $hasField -}}
+              Hooks: twirptrace.NewServerHooks(),
```

Comment:
There is the `ChainHooks` function that should be able to help in case the field is already set

Reply:
nice! didn't know about it